src/Microsoft.TestPlatform.Extensions.BlameDataCollector/Win32NamedEvent.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.Extensions.BlameDataCollector
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    public class Win32NamedEvent
+    {
+        private const uint EventModifyState = 0x0002;
+
+        private string eventName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Win32NamedEvent"/> class.
+        /// Create a NamedEvent object with the name of the event, and assume auto reset with
+        /// an initial state of reset.
+        /// </summary>
+        /// <param name="eventName">The name of the win32 event</param>
+        public Win32NamedEvent(string eventName)
+        {
+            this.eventName = eventName;
+        }
+
+        /// <summary>
+        /// Set the named event to a signaled state. The Wait() method will not block any
```

Comment:
> The Wait() method [](start = 52, length = 18)

What wait method are we refering to here ?

Reply:
oh this was from the msdn docs for this method. Need to go deeper into kernel events to understand :p